tests/conformance/bindings/bindings.go

```diff
@@ -0,0 +1,205 @@
+package bindings
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/dapr/components-contrib/bindings"
+	"github.com/dapr/components-contrib/tests/conformance/utils"
+	"github.com/dapr/dapr/pkg/logger"
+	"github.com/stretchr/testify/assert"
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+const (
+	defaultTimeoutDuration = 60 * time.Second
+)
+
+// nolint:gochecknoglobals
+var testLogger = logger.NewLogger("bindingsTest")
+
+type TestConfig struct {
+	utils.CommonConfig
+	inputMetadata      map[string]string
+	outputMetadata     map[string]string
+	readBindingTimeout time.Duration
+}
+
+func NewTestConfig(name string, allOperations bool, operations []string, config map[string]string) TestConfig {
+	testConfig := TestConfig{
+		CommonConfig: utils.CommonConfig{
+			ComponentType: "bindings",
+			ComponentName: name,
+			AllOperations: allOperations,
+			Operations:    sets.NewString(operations...),
+		},
+		inputMetadata:      make(map[string]string),
+		outputMetadata:     make(map[string]string),
+		readBindingTimeout: defaultTimeoutDuration,
+	}
+
+	for key, val := range config {
+		if key == "readBindingTimeout" {
+			timeout, err := strconv.Atoi(val)
+			if err == nil {
+				testConfig.readBindingTimeout = time.Duration(timeout) * time.Second
+			}
+		}
+
+		// A url indicates that we need to test the http binding which requires a simple endpoint.
+		if key == "url" {
+			startHTTPServer(val)
+		}
+
+		if strings.HasPrefix(key, "output_") {
+			testConfig.outputMetadata[strings.Replace(key, "output_", "", 1)] = val
+		}
+
+		if strings.HasPrefix(key, "input_") {
+			testConfig.inputMetadata[strings.Replace(key, "input_", "", 1)] = val
+		}
+	}
+
+	return testConfig
+}
+
+func startHTTPServer(url string) {
+	if parts := strings.Split(url, ":"); len(parts) != 2 {
+		testLogger.Errorf("URL must be in format {host}:{port}. Got: %s", url)
+	} else {
+		if port, err := strconv.Atoi(parts[1]); err != nil {
+			testLogger.Errorf("Could not parse port number: %s", err.Error())
+		} else {
+			go utils.StartHTTPServer(port)
```

Comment:
add functionality to stop go routine properly. Probably SIGTERM or a stop channel.

Reply:
Done.